pkg/cloudclient/gcp/private.go

```diff
@@ -115,10 +115,28 @@ func (c *Client) setDefaultAPIPublic(ctx context.Context, kclient client.Client,
 	return nil
 }
 
-func (c *Client) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
+func (c *GCPClient) ensureDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
 	// google.golang.org/api/dns/v1.Service is a struct, not an interface, which
 	// will make this all but impossible to write unit tests for
 
+	// Forwarding rule is necessary for rh-api lb setup
+	// Check forwarding rule exists first
+	if svc.Spec.Type == "LoadBalancer" {
+		ingressList := svc.Status.LoadBalancer.Ingress
+		if len(ingressList) == 0 {
+			// the LB doesn't exist
+			return cioerrors.NewLoadBalancerNotReadyError()
+		}
+		rhapiLbIP := ingressList[0].IP
```

Comment:
Will the target IP always be in the first item?

Reply:
I relied on Benson's existing logic to get hostname from the first item. This item has both hostname and IP. @bng0y  might know the answer to whether it's always the first item.